sessionctx/variable/sequence_state.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package variable
+
+import (
+	"sync"
+)
+
+// SequenceState cache all sequence's latest value accessed by lastval() builtins. It's a session scoped
+// variable, and all public methods of SequenceState are currently-safe.
+type SequenceState struct {
+	mu sync.Mutex
+	// latestValueMap caches the last value obtained by nextval() for each sequence id.
+	latestValueMap map[int64]int64
```

Comment:
Consider [sync.Map](https://golang.org/pkg/sync/#Map) and avoid use `sync.Mutex` manually?

Reply:
Maybe the `map[int64]int64` is more directly to know what kind of the k&v it is.
Sync.map will store the interface, and transform it, maybe too heavy for int64.